compose: Allow 3-way compose box resizing with new fixed expanded state.

N-Shar-ma opened the pull request:
Apart from the normal (collapsed) and full screen sizes, a new expanded state with the same size as the maximum a normal compose box can stretch to when full (40% of the screen height) is now available. Now a user can expand the compose box without it covering the full screen with a click. The vertical resize icon in the bottom right corner of the compose box is rendered useless so has been removed.

All three states can be cycled through by clicking the compose resize button in the order: collapsed -> 40% of the screen -> full screen. When a message naturally causes the compose box in its normal state to expand up to 40% of the screen, clicking the resize button will take it to full screen state.

Fixes: #29966.

**Screenshots and screen captures:**
![image](https://github.com/zulip/zulip/assets/68962290/abbdc9ea-4486-477f-9b89-d72285b382cc)

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [x] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

alya commented:
1. The tooltip should use "Maximize" rather than "Maximise", as that's standard in American English.
2. Expanding the compose box and then stretching my window (horizontally or vertically) produces a strange result:
![Screenshot 2024-06-12 at 08 39 12](https://github.com/zulip/zulip/assets/2090066/bb6d55ee-56bf-4b3d-b7ca-552afd288641)

alya commented:
More feedback [in #design](https://chat.zulip.org/#narrow/stream/101-design/topic/Implement.203-way.20compose.20box.20resizing.20.2329966/near/1824233).

N-Shar-ma commented:
> * The tooltip should use "Maximize" rather than "Maximise", as that's standard in American English.
> 
> * Expanding the compose box and then stretching my window (horizontally or vertically) produces a strange result:

Have addressed these comments but still working on feedback from the CZO design thread

N-Shar-ma commented:
Addressed all feedback, this is ready for more testing / review

(The error is related to testing and is being discussed [here](https://chat.zulip.org/#narrow/stream/49-development-help/topic/zjquery.20wont.20allow.20.60one.60.20and.20.60on.60.20handlers.20for.20same.20event))

alya commented:
Looks good to me in manual testing!

timabbott commented:
@N-Shar-ma can you DM me once you've rebased this one? I'm not sure how best to resolve the merge conflict.

timabbott commented:
Deploying on chat.zulip.org!

timabbott commented:
Merged, thanks for all the work on this transition @N-Shar-ma!
